File: _Src/Freem/Freem.Entities.UseCases.DTO/Activities/Update/UpdateActivityRequest.cs
using Freem.Entities.Activities.Identifiers;
using Freem.Entities.Activities.Models;
using Freem.Entities.Common.Relations.Collections;
using Freem.Entities.UseCases.DTO.Abstractions.Models;

namespace Freem.Entities.UseCases.DTO.Activities.Update;

public sealed class UpdateActivityRequest
{
    public ActivityIdentifier Id { get; }
    
    public UpdateField<ActivityName>? Name { get; init; }
    
    public UpdateField<RelatedTagsCollection>? Tags { get; init; }

    public UpdateActivityRequest(ActivityIdentifier id)
    {
        ArgumentNullException.ThrowIfNull(id);

        Id = id;
    }

    public bool HasChanges()
    {
        return Name is not null || Tags is not null;
    }
}

File: _Src/Freem/Freem.Web/src/clients/models/activities/UpdateActivityRequest.ts
import { UpdateField } from "../UpdateField";

export class UpdateActivityRequest {
  public readonly name?: UpdateField<string>;
  public readonly tags?: UpdateField<string[]>;

  constructor(name?: UpdateField<string>, tags?: UpdateField<string[]>) {
    this.name = name;
    this.tags = tags;
  }
}